src/map/map.cpp

```diff
@@ -17,19 +17,20 @@
 #include "io/iomap.hpp"
 #include "io/iomapserialize.hpp"
 
-bool Map::load(const std::string &identifier, const Position &pos, bool unload) {
+void Map::load(const std::string &identifier, const Position &pos) {
 	try {
-		IOMap::loadMap(this, identifier, pos, unload);
-		return true;
-	} catch (const IOMapException &e) {
-		g_logger().error("[Map::load] - {}", e.what());
-	} catch (const std::exception &) {
-		g_logger().error("[Map::load] - The map in folder {} is missing or corrupted", identifier);
+		path = identifier;
+		IOMap::loadMap(this, pos);
+	} catch (const std::exception &e) {
```

Comment:
This doesn't make sense to me, the exceptions can be multiple here, I'd just bubble them up and caught them in the thread. Making unrelated exceptions ios failure is a bit weird, imho.